drivers/sensor/bh1749/bh1749_trigger.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <gpio.h>
+#include <i2c.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <sensor.h>
+#include "bh1749.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(BH1749_TRIGGER, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Callback for active sense pin from BH1749 */
+static void bh1749_gpio_callback(struct device *dev, struct gpio_callback *cb,
+				 u32_t pins)
+{
+	struct bh1749_data *drv_data =
+		CONTAINER_OF(cb, struct bh1749_data, gpio_cb);
+
+	gpio_pin_disable_callback(dev, DT_ROHM_BH1749_0_INT_GPIOS_PIN);
+
+	k_work_submit(&drv_data->work);
+}
+
+static void bh1749_work_cb(struct k_work *work)
+{
+	struct bh1749_data *data = CONTAINER_OF(work,
+						struct bh1749_data,
+						work);
+	struct device *dev = data->dev;
+
+	if (data->trg_handler != NULL) {
+		data->trg_handler(dev, &data->trigger);
+	}
+}
+
+/* Set sensor trigger attributes */
+int bh1749_attr_set(struct device *dev,
+		    enum sensor_channel chan,
+		    enum sensor_attribute attr,
+		    const struct sensor_value *val)
+{
+	struct bh1749_data *data = dev->driver_data;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	if (attr == SENSOR_ATTR_UPPER_THRESH) {
+		if (i2c_reg_write_byte(data->i2c,
+				       DT_ROHM_BH1749_0_BASE_ADDRESS,
+				       BH1749_TH_HIGH_LSB,
+				       (u8_t)val->val1)) {
+			LOG_ERR("Could not set threshold\r\n");
```

Comment:
No need for this extra "\r\n" (same goes for all error messages in this function).

I think it would be good to differentiate the error messages logged by this function and use "Could not set upper threshold LSB" here, for instance.